### PR TITLE
FreeRTOS_DNS : changes to make asynchronous lookup work, version 3

### PR DESCRIPTION
**Pull request by htibosch**
Description
-----------
This PR replaces the older [PR 860](https://github.com/aws/amazon-freertos/pull/860), which had merging conflicts.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by markrtuttle** [comment]
Patch contained in https://github.com/markrtuttle/amazon-freertos/tree/tibosch1-tuttle

File: libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c [hunk 1]
```diff
@@ -378,126 +414,146 @@ typedef struct xDNSAnswerRecord DNSAnswerRecord_t;
 	/*-----------------------------------------------------------*/
 
 	/* A DNS reply was received, see if there is any matching entry and
-	call the handler. */
-	static void vDNSDoCallback( TickType_t xIdentifier, const char *pcName, uint32_t ulIPAddress );
-	static void vDNSDoCallback( TickType_t xIdentifier, const char *pcName, uint32_t ulIPAddress )
+	call the handler.  Returns pdTRUE if uxIdentifier was recognised. */
+	static BaseType_t xDNSDoCallback( TickType_t uxIdentifier,
+									  const char *pcName,
+									  uint32_t ulIPAddress );
```
Comment (dan4thewin) [inline libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c]: Is there any value to having a prototype immediately above it's function?

Comment (htibosch) [reply libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c]: It makes the code look less complicated.  The code for asynchronous callbacks is gathered in a big block :

~~~c
#if( ipconfigDNS_USE_CALLBACKS == 1 )
    /* 166 lines of code handling call-backs */
#endif
~~~

It is old code, and it is against the style rule that an `#ifdef` should include a single function.
In this block, a entire feature is included.
I don't mind to change this in a separate PR.

Hein

**Comment by htibosch** [comment]
Closing this PR for now, it has been replaced with [#1415](https://github.com/aws/amazon-freertos/pull/1415)

Hein

**Pull request closed without merging**